test/test_reads.py

```diff
@@ -263,6 +263,49 @@ def testWalkHSPLeftAndRightOverhangingMatch(self):
                           (13, 'T', False)],
                          list(read.walkHSP(hsp)))
 
+    def testcheckAlphabetwithReadMustBePermissive(self):
+        """
+        The checkAlphabet function must be permissive if a dark.Read is
+        passed.
+        """
+        read = Read('id', 'ARSTGATGCASASASASASAS')
+        self.assertEqual(set('ACGSRT'), read.checkAlphabet())
+
+    def testcheckAlphabetAAReadMatchingReturnTrue(self):
+        """
+        If an AA read with an AARead readClass is passed in, the checkAlphabet
+        function must return True.
+        """
+        read = AARead('id', 'ARSTGATGCASASASASASAS')
+        self.assertEqual(set('ACGSRT'), read.checkAlphabet())
+
+    def testcheckAlphabetDNAReadMatchingReturnTrue(self):
+        """
+        If a DNA read with a DNARead readClass is passed in, the checkAlphabet
+        function must return True.
+        """
+        read = DNARead('id', 'AAATTAACGGGCCTAGG')
+        self.assertEqual(set('ACTG'), read.checkAlphabet())
+
+    def testcheckAlphabetAAReadNotMatchingRaise(self):
+        """
+        If an AA read with a DNARead readClass is passed in, the checkAlphabet
+        function must raise an IndexError.
+        """
+        read = AARead('id', 'AAATTAACGGGCCTAGG')
+        error = ("It looks like a DNA sequence has been passed to AARead().")
+        self.assertRaisesRegexp(ValueError, error, read.checkAlphabet)
+
+    def testcheckAlphabetDNAReadNotMatchingRaise(self):
+        """
+        If a DNA read with an AARead readClass is passed in, the checkAlphabet
+        function must raise an IndexError.
+        """
+        read = DNARead('id', 'ARSTGATGCASASASASASAS')
+        error = ("Read alphabet \(\'ACGRST\'\) is not a subset of expected "
```

Comment:
Oh shit, maybe I'm wrong. I thought `len("\(")` would be one, but it's 2.  So the backslash is making it through to the regex. Sorry!